server/stream.go

```diff
@@ -71,6 +74,12 @@ type StreamConfig struct {
 	AllowRollup bool `json:"allow_rollup_hdrs"`
 }
 
+// SubjectMapping allows a source subject to be mapped to a destination subject for republishing.
+type SubjectMapping struct {
+	Source      string `json:"src,omitempty"`
```

Comment:
Is this really optional? not setting it just gives me `stream configuration for republish source is not valid subset of subjects`